core/src/SYCL/Kokkos_SYCL_ParallelFor_Range.hpp

```diff
@@ -94,29 +106,31 @@ class Kokkos::Impl::ParallelFor<FunctorType, Kokkos::RangePolicy<Traits...>,
       (void)memcpy_event;
 #endif
 
+      const auto actual_range = policy.end() - policy.begin();
       if (policy.chunk_size() <= 1) {
 #ifdef SYCL_EXT_ONEAPI_AUTO_LOCAL_RANGE
-        const auto actual_range = policy.end() - policy.begin();
         FunctorWrapperRangePolicyParallelForCustom<Functor, Policy> f{
             policy.begin(), functor, actual_range};
         // Round the actual range up to the closest power of two not exceeding
         // the maximum workgroup size
-        const auto max_wgroup_size =
+        const std::size_t max_wgroup_size =
             q.get_device().get_info<sycl::info::device::max_work_group_size>();
-        const auto wgroup_size_multiple = Kokkos::bit_floor(
+        const std::size_t wgroup_size_multiple = Kokkos::bit_floor(
             std::min<std::size_t>(max_wgroup_size, actual_range));
 
-        const auto launch_range = (actual_range + wgroup_size_multiple - 1) /
-                                  wgroup_size_multiple * wgroup_size_multiple;
+        const std::size_t launch_range =
+            (actual_range + wgroup_size_multiple - 1) / wgroup_size_multiple *
+            static_cast<size_t>(wgroup_size_multiple);
         sycl::nd_range<1> range(
-            launch_range, sycl::ext::oneapi::experimental::auto_range<1>());
+            std::min<std::size_t>(launch_range, INT_MAX),
```

Comment:
Why are we "clamping" to `INT_MAX`?
Are we working around a bug in the SYCL implementation?  It looks like `sycl::range` constructs with `size_t` and there was no precondition on a maximum value in the doc https://github.khronos.org/SYCL_Reference/iface/range.html#range

Reply:
This is one step into not requiring to force https://www.intel.com/content/www/us/en/docs/dpcpp-cpp-compiler/developer-guide-reference/2023-2/fsycl-id-queries-fit-in-int.html, and yes, https://github.com/intel/llvm/issues/4255 is an issue decribing problems with ranges larger than `INT_MAX`.